src/service-broker.js

```diff
@@ -986,36 +986,85 @@ class ServiceBroker {
 	waitForServices(serviceNames, timeout, interval, logger = this.logger) {
 		if (!Array.isArray(serviceNames)) serviceNames = [serviceNames];
 
-		serviceNames = _.uniq(
+		class ServiceToken {
```

Comment:
I feel the PR is a little complicated. Please try to solve it without sub-classes. Regarding performance, it's not optimal to create sub-classes and instances of classes in a method.

Reply:
Class based was the cleanest/simplest solution I found working on this - I toyed with a pure object/function but that got out of hand imo. Do you have any suggestions for a different approach? Thanks for your time here

Reply:
@icebob Would you be comfortable moving forward with this approach if we found a better place to store these classes?